src/scf-release/src/acceptance-tests-brain/test-scripts/018_autoscaler_test.rb

```diff
@@ -0,0 +1,143 @@
+                 'autoscaler-postgres',
+                 'autoscaler-scalingengine',
+                 'autoscaler-scheduler',
+                 'autoscaler-servicebroker' ]
+
+# ~ ~~ ~~~ ~~~~~ ~~~~~~~~ ~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
+## Check if autoscaler pods are running
+
+have = 0
+STATEFULSETS.each do |name|
+  have += 1 if statefulset_ready(NAMESPACE, name)
+end
+
+# Skip the test if none of the autoscaler pods are running.
+puts "Autoscaler inactive." if have == 0
+exit_skipping_test if have == 0
+
+# Fail the test if some, but not all of the autoscaler pods are
+# running.
+fail "Autoscaler only partially active." if have < STATEFULSETS.length
+
+puts "Autoscaler active. Testing begins."
+STDOUT.flush
+
+# ~ ~~ ~~~ ~~~~~ ~~~~~~~~ ~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
+## Standard cf cli access
+
+login
+setup_org_space
+
+# ~ ~~ ~~~ ~~~~~ ~~~~~~~~ ~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
+##
+
+dora         = resource_path('dora')
+app_name     = random_suffix('dora')
+broker_name  = random_suffix('SCALER')
+service_name = random_suffix('SERVICE')
+url          = "http://#{app_name}.#{APPS_DOMAIN}"
+
+at_exit do
+  puts "Exiting. Cleanup."
+  STDOUT.flush
+  set errexit: false do
+    run "cf", "unbind-service", app_name, service_name
+    run "cf", "delete", "-f", app_name
+    run "cf", "delete-service", service_name, "-f"
+    run "cf", "delete-service-broker", broker_name, "-f"
+    end
+end
+
+run "cf", "push", app_name, "--no-start", "-p", dora
+run "cf", "create-service-broker", broker_name, BROKER_USER, BROKER_PASS, BROKER_URL
+run "cf", "service-access"
+run "cf", "enable-service-access", SMOKE_SERVICE, "-p", SMOKE_PLAN
+run "cf", "create-service", SMOKE_SERVICE, SMOKE_PLAN, service_name
+
+run "cf", "bind-service", app_name, service_name, "-c", "{
+    \"instance_min_count\": 1,
+    \"instance_max_count\": 4,
+    \"scaling_rules\": [{
+        \"metric_type\": \"memoryused\",
+        \"stat_window_secs\": 60,
+        \"breach_duration_secs\": 60,
+        \"threshold\": 10,
+        \"operator\": \">=\",
+        \"cool_down_secs\": 300,
+        \"adjustment\": \"+1\"
+    }]
+}"
+run "cf", "start", app_name
+
+puts "Waiting for the app to start..."
+STDOUT.flush
+run_with_retry 120, 1 do
+  run "curl --silent '#{url}/' | grep Dora"
+end
+puts "\nApp started.\n"
+STDOUT.flush
+
+# ~ ~~ ~~~ ~~~~~ ~~~~~~~~ ~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
+
+$app_uuid = capture "cf", "app", "--guid", app_name
+
+def get_count
+  capture("cf curl /v2/apps/#{$app_uuid} | jq .entity.instances").to_i
+end
+
+puts "Checking that we initially have one instance...\n"
+STDOUT.flush
+
+run "cf", "app", app_name
+count = get_count
+fail "App has #{count} instances!\n" if count != 1
+
+puts "Causing memory stress...\n"
+STDOUT.flush
+run "curl", "-X", "POST", "#{url}/stress_testers?vm=10&vm-bytes=100M"
+
+puts "Waiting for new instances to start..."
+STDOUT.flush
+run_with_retry 24, 5 do
+  if get_count == 1
+    puts "."
```

Comment:
I don't follow this `.` (and I like punctuation).

Reply:
Also from the original code. ... And bogus for brain tests.

In the original shell script you would see a growing line of dots as it keeps checking.
In the brain test the logging is different, so this can go.

Reply:
Done in commit a36b11ea4c26c48873d3061aea8cd416486b68c7